app/Listeners/UpdateProfilePicture.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Listeners;
+
+use App\Events\ProfilePictureUpdated;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Facades\Storage;
+use Intervention\Image\Facades\Image;
+
+class UpdateProfilePicture implements ShouldQueue
+{
+    /**
+     * Create the event listener.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * Handle the event.
+     *
+     * @param  ProfilePictureUpdated  $event
+     * @return void
+     */
+    public function handle(ProfilePictureUpdated $event)
+    {
+        $filename = time() . '.' . $event->image_ext;
```

Comment:
Could we instead use `$image->hashName()` so we don't have any possible collisions?